internal/agent/transport/grpc.go

```diff
@@ -0,0 +1,170 @@
+package transport
+
+import (
+	"context"
+	"errors"
+	"io"
+
+	"github.com/avast/retry-go"
+	"github.com/go-logr/logr"
+	"github.com/tinkerbell/tink/internal/agent/event"
+	"github.com/tinkerbell/tink/internal/agent/workflow"
+	workflowproto "github.com/tinkerbell/tink/internal/proto/workflow/v2"
+)
+
+var _ event.Recorder = &GRPC{}
+
+func NewGRPC(log logr.Logger, client workflowproto.WorkflowServiceClient) *GRPC {
+	return &GRPC{
+		log:    log,
+		client: client,
+	}
+}
+
+type GRPC struct {
+	log    logr.Logger
+	client workflowproto.WorkflowServiceClient
+}
+
+func (g *GRPC) Start(ctx context.Context, agentID string, handler WorkflowHandler) error {
```

Comment:
hmmm.....we dont have a linting check for code comments on exported functions/methods?

Reply:
Apparently not 🤷🏻 . We need to tweak the `ci-scripts` stuff and align Tinkerbell repositories. As always, a low priority task.